pkg/infra/pulumi_aws/deploylib.ts

```diff
@@ -1365,30 +1365,44 @@ export class CloudCCLib {
             Action: ['ssm:GetParameters', 'secretsmanager:GetSecretValue', 'kms:Decrypt', 'ecr:*'],
             Resource: '*',
         })
-
-        const nlb = new awsx.lb.NetworkLoadBalancer(`${execUnitName}-nlb`, {
-            external: false,
-            vpc: this.klothoVPC,
-            subnets: this.privateSubnetIds,
+        let needsLoadBalancer = false
+        this.topology.topologyIconData.forEach((resource) => {
+            if (resource.kind === Resource.gateway) {
+                this.topology.topologyEdgeData.forEach((edge) => {
+                    if (edge.source == resource.id && edge.target === `${execUnitName}_exec_unit`) {
```

Comment:
nit, unless this is intentional (in which case I'd ask for a comment):

```suggestion
                    if (edge.source === resource.id && edge.target === `${execUnitName}_exec_unit`) {
```